airflow/migrations/versions/3c20cacc0044_add_dagrun_run_type.py

```diff
@@ -0,0 +1,58 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Add DagRun run_type
+
+Revision ID: 3c20cacc0044
+Revises: 952da73b5eff
+Create Date: 2020-04-08 13:35:25.671327
+
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+from airflow.models import DagRun
+from airflow.utils.types import DagRunType
+
+# revision identifiers, used by Alembic.
+revision = "3c20cacc0044"
+down_revision = "952da73b5eff"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    """Apply Add DagRun run_type"""
+    op.add_column("dag_run", sa.Column("run_type", sa.String(length=50), nullable=True))
+
+    connection = op.get_bind()
+    sessionmaker = sa.orm.sessionmaker()
+    session = sessionmaker(bind=connection)
+
+    for run_type in DagRunType:
+        session.query(DagRun).filter(DagRun.run_id.like(f"{run_type.value}__%")).update(
+            {DagRun.run_type: run_type.value}, synchronize_session=False
+        )
```

Comment:
Run_id can be anything in case of a manual run, so I think we also need a `SET run_type = 'manual' where run_type is null` afterwards.

Reply:
I don't see this done anywhere ^^

Reply:
This needs doing

Reply:
Done